### PR TITLE
Upgrade to Spring Boot 3.0

### PR DESCRIPTION
**Pull request by mraible**
Related to https://github.com/jhipster/generator-jhipster/pull/19791.

### CODE REVIEW

**Comment by mraible** [comment]
@DanielFran Do you know how to fix the JDK issue? I can't seem to figure out where to set Java 17.

```
[INFO] --- maven-enforcer-plugin:3.1.0:enforce (enforce-versions) @ jhipster-parent ---
Error:  Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
You are running an older version of Java. JHipster requires at least JDK 17
```

**Comment by DanielFran** [comment]
> @DanielFran Do you know how to fix the JDK issue? I can't seem to figure out where to set Java 17.
> 
> ```
> [INFO] --- maven-enforcer-plugin:3.1.0:enforce (enforce-versions) @ jhipster-parent ---
> Error:  Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
> You are running an older version of Java. JHipster requires at least JDK 17
> ```

@mraible This happens compiling the bom only, or the generator?

**Comment by mraible** [comment]
@DanielFran Marcelo fixed the compilation issue.

**Comment by mraible** [comment]
I was unable to find an `OpenApiCustomer` class after upgrading to SpringDoc 2.0.0-M6. Maybe it's not needed anymore?

File: jhipster-framework/src/main/java/tech/jhipster/config/apidoc/JHipsterSpringDocGroupsConfiguration.java [hunk 1]
```diff
@@ -94,7 +96,7 @@ public JHipsterOpenApiCustomizer jhipsterOpenApiCustomizer() {
     @Bean
     @ConditionalOnMissingBean(name = "openAPIDefaultGroupedOpenAPI")
     public GroupedOpenApi openAPIDefaultGroupedOpenAPI(
-        List<OpenApiCustomiser> openApiCustomisers,
+        List<OpenApiCustomizer> openApiCustomisers,
```
Comment (DanielFran) [inline jhipster-framework/src/main/java/tech/jhipster/config/apidoc/JHipsterSpringDocGroupsConfiguration.java:99]:
```suggestion
        List<OpenApiCustomizer> openApiCustomizers,
```

Comment (DanielFran) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/JHipsterSpringDocGroupsConfiguration.java:99]: Rename variable

**Review by DanielFran — CHANGES_REQUESTED** [review]
Rename variable

**Comment by mshima** [comment]
Can we change the version to don't conflict with current main?

**Comment by mraible** [comment]
Yes, feel free to change the version.

**Comment by mshima** [comment]
8.0.0-SNAPSHOT?

**Comment by mraible** [comment]
Works for me.

**Comment by mraible** [comment]
I'm able to reproduce the failing e2e test that happens with the ng-default job. It seems that when you edit the field, there's two required fields that need to be selected before the save button is enabled.

```
[1]   TestTwoRelationshipsSameEntity e2e test
[1]     ✓ TestTwoRelationshipsSameEntities menu should load TestTwoRelationshipsSameEntities page (1230ms)
[1]     TestTwoRelationshipsSameEntity page
[1]       create button click
[1]         ✓ should load create TestTwoRelationshipsSameEntity page (649ms)
[1]       with existing value
[1]         ✓ detail button click should load details TestTwoRelationshipsSameEntity page (636ms)
[1]         ✓ edit button click should load edit TestTwoRelationshipsSameEntity page and go back (608ms)
[1]         (Attempt 1 of 3) edit button click should load edit TestTwoRelationshipsSameEntity page and save
[1]         (Attempt 2 of 3) edit button click should load edit TestTwoRelationshipsSameEntity page and save
[1]         1) edit button click should load edit TestTwoRelationshipsSameEntity page and save
[1]         ✓ last delete button click should delete instance of TestTwoRelationshipsSameEntity (676ms)
[1]     new TestTwoRelationshipsSameEntity page
[1] Cannot find coverage file /home/runner/work/generator-jhipster/generator-jhipster/app/.nyc_output/out.json
[1] Skipping coverage report
[1]       ✓ should create an instance of TestTwoRelationshipsSameEntity (837ms)
[1] 
[1] 
[1]   6 passing (21s)
[1]   1 failing
[1] 
[1]   1) TestTwoRelationshipsSameEntity e2e test
[1]        TestTwoRelationshipsSameEntity page
[1]          with existing value
[1]            edit button click should load edit TestTwoRelationshipsSameEntity page and save:
[1]      CypressError: Timed out retrying after 4050ms: `cy.click()` failed because this element is `disabled`:
[1] 
[1] `<button type="submit" id="save-entity" data-cy="entityCreateSaveButton" class="btn btn-primary" disabled="">...</button>`
```

File: jhipster-dependencies/pom.xml [hunk 2]
```diff
@@ -51,32 +51,27 @@
...
-        <problem-spring-web.version>${problem-spring.version}</problem-spring-web.version>
-        <problem-spring-webflux.version>${problem-spring.version}</problem-spring-webflux.version>
-        <prometheus-simpleclient.version>0.15.0</prometheus-simpleclient.version>
         <redisson.version>3.18.1</redisson.version>
-        <reflections.version>0.10.1</reflections.version>
-        <spring-cloud.version>2021.0.5</spring-cloud.version>
-        <spring-cloud-netflix.version>2.2.10.RELEASE</spring-cloud-netflix.version>
-        <spring-cloud-security.version>2.2.5.RELEASE</spring-cloud-security.version>
+        <reflections.version>0.10.2</reflections.version>
+        <spring-cloud.version>2022.0.0-RC2</spring-cloud.version>
```
Comment (mraible) [inline jhipster-dependencies/pom.xml:71]:
```suggestion
        <spring-cloud.version>2022.0.0-RC3</spring-cloud.version>
```
https://spring.io/blog/2022/12/02/spring-cloud-2022-0-0-release-candidate-3-codename-kilburn-has-been-released

**Pull request merged**